aztk/utils/azure_api.py

```diff
@@ -16,80 +16,47 @@
-        self.resource_id = resource_id
 
+def validate_secrets(secrets):
+    if secrets.service_principal:
+        if not RESOURCE_ID_PATTERN.match(secrets.service_principal.batch_account_resource_id):
+            raise error.AzureApiInitError("ServicePrincipal batch_account_resource_id is not in expected format")
 
-def _validate_batch_config(batch_config: BatchConfig):
-    if batch_config.resource_id is None:
-        if batch_config.account_name is None or batch_config.account_key is None or batch_config.service_url is None:
-            raise error.AzureApiInitError("Neither servicePrincipal nor sharedKey configured for Batch in secrets.yaml config")
-    else:
-        if not RESOURCE_ID_PATTERN.match(batch_config.resource_id):
-            raise error.AzureApiInitError("servicePrincipal.batchaccountresourceid is not in expected format")
-        if batch_config.tenant_id is None:
-            raise error.AzureApiInitError("Missing tenant_id in servicePrincipal auth for Batch in secrets.yaml config")
-        if batch_config.client_id is None:
-            raise error.AzureApiInitError("Missing client_id in servicePrincipal auth for Batch in secrets.yaml config")
-        if batch_config.credential is None:
-            raise error.AzureApiInitError("Missing credential in servicePrincipal auth for Batch in secrets.yaml config")
+        if not RESOURCE_ID_PATTERN.match(secrets.service_principal.storage_account_resource_id):
+            raise error.AzureApiInitError("ServicePrincipal storage_account_resource_id is not in expected format")
 
 
-def make_batch_client(batch_config: BatchConfig):
+def make_batch_client(secrets):
     """
         Creates a batch client object
         :param str batch_account_key: batch account key
         :param str batch_account_name: batch account name
         :param str batch_service_url: batch service url
     """
     # Validate the given config
-    _validate_batch_config(batch_config)
+    credentials = None
 
-    if batch_config.resource_id is None:
+    if secrets.shared_key:
         # Set up SharedKeyCredentials
-        base_url = batch_config.service_url
+        base_url = secrets.shared_key.batch_service_url
         credentials = batch_auth.SharedKeyCredentials(
-            batch_config.account_name,
-            batch_config.account_key)
+            secrets.shared_key.batch_account_name,
+            secrets.shared_key.batch_account_key)
     else:
         # Set up ServicePrincipalCredentials
-        credentials = ServicePrincipalCredentials(
-            client_id=batch_config.client_id,
-            secret=batch_config.credential,
-            tenant=batch_config.tenant_id,
+        arm_credentials = ServicePrincipalCredentials(
+            client_id=secrets.service_principal.client_id,
+            secret=secrets.service_principal.credential,
+            tenant=secrets.service_principal.tenant_id,
             resource='https://management.core.windows.net/')
```

Comment:
Is it OK to hard code this url (and the batch one below)? Are there any situations where a user might need to change them (e.g. maybe China)?